src/wrap.js

```diff
@@ -11,31 +14,85 @@ import { useImperativeHandle, forwardRef } from 'react';
 // More information on `useImperativeHandle`:
 // <https://reactjs.org/docs/hooks-reference.html#useimperativehandle>
 //
-// functionComponent - The function component you want to test.
-//
 // Example
 //
-//   import { Button } from 'react-native-elements';
-//   import { hook, wrap } from 'cavy';
-//
-//   class MyComponent extends React.Component {
-//     // ...
-//     render() {
-//       const WrappedButton = wrap(Button);
-//
-//       return (
-//         <WrappedButton ref={this.generateTestHook('button')} onPress={}/>
-//       )
-//     }
-//   }
-//   export default hook(MyComponent);
-//
-export default function wrap(functionComponent) {
-  // `forwardRef` accepts a render function that receives our props and ref.
-  return forwardRef((props, ref) => {
-    // It returns the wrapped component after calling `useImperativeHandle`, so
-    // that our ref can be used to call the inner function component's props.
-    useImperativeHandle(ref, () => ({ props }));
-    return functionComponent(props);
-  });
+// import React from 'react';
+// import { Button } from 'react-native-elements';
+// import { useCavy, wrap } from 'cavy';
+//
+// export default () => {
+//   const generateTestHook = useCavy();
+//   const TestableButton = wrap(Button);
+//
+//   return (
+//     <TestableButton ref={this.generateTestHook('button')} onPress={}/>
+//   )
+// };
+//
+//
+//
+// With a component like `Text`:
+//
+// Higher-order component that wraps a component like `Text`, which is neither
+// a React Class nor a Function Component, and returns a React Class with
+// testable props.
+//
+// Example:
+//
+// import React from 'react';
+// import { View, Text } from 'react-native';
+// import { useCavy, wrap } from 'cavy';
+//
+// export default ({ data }) => {
+//   const generateTestHook = useCavy();
+//   const TestableText = wrap(Text);
+//
+//   return (
+//     <View>
+//       <TestableText ref={generateTestHook('title')}>
+//         {data.title}
+//       </TestableText>
+//     </View>
+//   )
+// };
+//
+export default function wrap(Component) {
+  if (typeof Component === 'function' && isNotReactClass(Component)) {
+    // `forwardRef` accepts a render function that receives our props and ref.
+    return forwardRef((props, ref) => {
+      // It returns the wrapped component after calling `useImperativeHandle`, so
+      // that our ref can be used to call the inner function component's props.
+      useImperativeHandle(ref, () => ({ props }));
+      return Component(props);
+    });
+  }
+
+  if (typeof Component == 'object') {
+    class WrapperComponent extends React.Component {
```

Comment:
Is this really the minimum we have to do to wrap it? We don't have to hoist statics or anything weird like that?

(sorry if this is a silly question)

Reply:
> Happy with the name `containsText` as it's nice and friendly.

👌 

> API all looks fine, so feel free to start updating the documentation too.

Will do!

> Dang, we could really do with #147. I got stuck in that PR getting the build working, I wonder if it would be easier now we've got the sample app more up-to-date? 🤷

Yeah I thought this too as I wanted to test passing in class components into `wrap` i.e. incorrect use-cases. I'll try building it today, but I'd like to get this in regardless.

> I can see a test for `wrap(Text)`, but do we test that `wrap(FunctionComponent)` works anywhere in the sample app? [This ActionButton test?](https://github.com/pixielabs/cavy/blob/extend-wrap/sample-app/CavyDirectory/app/ActionBar.js#L33) If so, I'm happy for this to go in.

We test that the button is there `await spec.exists('ActionBar.EmailButton')` - I'll add one that interacts with the button's props.

> Have you thought about the version bump? I don't think we'll break anyone's code with this unless they were doing some funky workaround, so... minor bump?

I was thinking minor bump. If people were working around this, then their workarounds should still be valid.

Reply:
Have hoisted statics.